### PR TITLE
PLF-8405 : remove double cote when searching for a document.

### PR DESCRIPTION
**Pull request by waelghannem**
remove double cote when searching for a document.

### CODE REVIEW

File: core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/control/UIAddressBar.java [hunk 1]
```diff
@@ -273,9 +274,10 @@ public void execute(Event<UIAddressBar> event) throws Exception {
       }else {
         queryStatement = StringUtils.replace(SQL_QUERY,"$0",currentNode.getPath());
       }
-      String escapedText = org.exoplatform.services.cms.impl.Utils.escapeIllegalCharacterInQuery(text);
+      text = text.replace("\"", "\\\"");
```
Comment (boubaker) [inline core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/control/UIAddressBar.java]: this change should be made in a centralized way inside `org.exoplatform.services.wcm.search.base.PageListFactory.searchInJCR(String, String, String, boolean, NodeSearchFilter, SearchDataCreator<E>, QueryCriteria)`

Comment (AzmiTouil) [reply core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/control/UIAddressBar.java]: Done

File: core/search/src/main/java/org/exoplatform/services/wcm/search/base/PageListFactory.java [hunk 2]
```diff
@@ -204,6 +204,11 @@
     QueryManager queryManager = session.getWorkspace().getQueryManager();
     Query query = queryManager.createQuery(queryStatement, language);
 
+    String keyword = criteria.getKeyword();
```
Comment (boubaker) [inline core/search/src/main/java/org/exoplatform/services/wcm/search/base/PageListFactory.java]: does this fix the problem ? I don't see where the `keyword` is used after.

Comment (AzmiTouil) [reply core/search/src/main/java/org/exoplatform/services/wcm/search/base/PageListFactory.java]: yes the problem was fixed,I change the keyword in the criteria, and the criteria will be used after

Comment (boubaker) [reply core/search/src/main/java/org/exoplatform/services/wcm/search/base/PageListFactory.java]: > will be used after

where, could you be more specific please

Comment (AzmiTouil) [reply core/search/src/main/java/org/exoplatform/services/wcm/search/base/PageListFactory.java]: That was my mistake, it works but the result comes from the search in ElasticSearch(searchInES) not in jcr.
To check

File: core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/control/UIAddressBar.java [hunk 3]
```diff
@@ -273,11 +273,20 @@ public void execute(Event<UIAddressBar> event) throws Exception {
       }else {
         queryStatement = StringUtils.replace(SQL_QUERY,"$0",currentNode.getPath());
       }
-      String escapedText = org.exoplatform.services.cms.impl.Utils.escapeIllegalCharacterInQuery(text);
-      queryStatement = StringUtils.replace(queryStatement,"$1", escapedText);
-      queryStatement = StringUtils.replace(queryStatement,"$2", escapedText.toLowerCase());
-      queryStatement = StringUtils.replace(queryStatement,"$3", escapedText.toLowerCase());
-      queryStatement = StringUtils.replace(queryStatement,"$4", escapedText.toLowerCase());
+      if (text.startsWith("\"") && text.endsWith("\"")) {
```
Comment (boubaker) [inline core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/control/UIAddressBar.java]: What if the user uses a search with the following exact term:
`"test search sentence" test word`

**Pull request closed without merging**